src/autopas/containers/directSum/DirectSum.h

```diff
@@ -117,6 +117,22 @@ class DirectSum : public CellBasedParticleContainer<FullParticleCell<Particle>>
     traversal->endTraversal();
   }
 
+  void iterateTriwise(TraversalInterface<InteractionTypeOption::threeBody> *traversal) override {
+    // Check if traversal is allowed for this container and give it the data it needs.
+    auto *traversalInterface = dynamic_cast<DSTraversalInterface<ParticleCell> *>(traversal);
+    auto *cellPairTraversal = dynamic_cast<CellTraversal<ParticleCell> *>(traversal);
+    if (traversalInterface && cellPairTraversal) {
+      cellPairTraversal->setCellsToTraverse(this->_cells);
+    } else {
+      autopas::utils::ExceptionHandler::exception(
+          "trying to use a traversal of wrong type in DirectSum::iterateTriwise");
+    }
```

Comment:
This is no different from the check in `iteratePairwise`. My understanding is that the `cellPairTraversal` is just a misnamed variable. But regardless, I'm not sure this actually checks that a traversal is triwise - you check that it is a DirectSum traversal and that it is a cell traversal.

Reply:
The "check" whether the traversal is pairwise or triwise already comes from the function signature.